src/anm/events.js

```diff
@@ -164,28 +164,31 @@ MouseEventsSupport.prototype.dispatch = function(event) {
         return true;
     }
 
-    var dispatchedChild;  // found the matching child inside
-    if (owner.inside(localEvent)) {
-        owner.reverseEach(function(child) {
-            if (child.isActive()) {
-                dispatchedChild = child.dispatchMouseEvent(localEvent) ? child : null;
-                if (dispatchedChild) return false; // stop iteration of reverseEach
-            }
-        });
+    var dispatchedByOwner, // handled event myself
+        dispatchedByChild; // not handled myself, but found the matching child inside
 
-        if (!dispatchedChild) {
-            if (localEvent.type === 'mousemove') {
-                this.processMove(localEvent); // fire mouseenter/mouseexit if required
-            }
-            if ((localEvent.type === 'mouseclick') || (localEvent.type === 'mousedown')) {
-                state.pressedNode = this.owner;
-            }
-            this.owner.fire(localEvent.type, localEvent);
+    // here and below localEvent has properties `.x` and `.y`, so duck typing works
+    dispatchedByOwner = owner.inBounds && owner.inBounds(localEvent) && owner.inside && owner.inside(localEvent);
```

Comment:
`contains` is relevant both for matrices and curves in real-life case, like we have here, and that's why writing a spec takes two days and implementing it in real life takes more. `transformToChild` === `element.adapt`, `contains` === `element.inside` (sorry they were named differently, but it happened before the spec), both are already here. In your spec `contains` is *not* checking children, so I suppose it's the answer to Dima -> it should not iterate through children.